### ISSUE TITLE
claymore v11...

### ISSUE THREAD

**Issue by PeshBG** [Demion/nodevfee#28]
It is not working with v11

**Comment by Demion** [comment]
It is working with Claymore **v11.0**.

https://i.imgur.com/P59wCCz.png
https://i.imgur.com/HrmrC6C.png
```
s = 0x02C8 flags = 0x0000 len =  148 buf = 7B 22 77 6F 72 6B 65 72 22 3A 20 22 65 74 68 31 2E 30 22 2C 20 22 6A 73 6F 6E 72 70 63 22 3A 20 22 32 2E 30 22 2C 20 22 70 61 72 61 6D 73 22 3A 20 5B 22 30 78 63 62 34 65 66 66 64 65 62 34 36 34 37 39 63 61 61 30 66 65 66 35 66 35 65 33 35 36 39 65 34 38 35 32 66 37 35 33 61 32 2E 77 6F 72 6B 65 72 31 22 2C 20 22 78 22 5D 2C 20 22 69 64 22 3A 20 32 2C 20 22 6D 65 74 68 6F 64 22 3A 20 22 65 74 68 5F 73 75 62 6D 69 74 4C 6F 67 69 6E 22 7D 0A 
{"worker": "eth1.0", "jsonrpc": "2.0", "params": ["0xcb4effdeb46479caa0fef5f5e3569e4852f753a2.worker1", "x"], "id": 2, "method": "eth_submitLogin"}
```
Although, actual mining efficiency you need test by yourself.

> Users report some actual efficiency decrease starting from version 10.1 and later (might be some hidden protection). Seems to work fine on 9.7 and ethermine for me.

For further troubleshooting upload your Claymore Log and `nodevfeeLog.txt`.

**Comment by babbage2018** [comment]
[nodevfeeLog.txt](https://github.com/Demion/nodevfee/files/1724374/nodevfeeLog.txt)  work работает

**Comment by br1egel** [comment]
well done, are you able to look in to the claymore.exe v11 ?
I deobfuscated this https://bitcointalk.org/index.php?topic=2274616.0 
hack. the guy did a different approach and used kernel32.dll to actually write the registers used in claymores version 10.0 by ReadProcessMemory and WriteProcessMemory. Unfortunately, the guy that wrote the hack also incorporated his wallet address. dll was gzipped and obfuscated with .net reactor 5.0. 
After a lot of debugging the claymore version 11, I found some addresses in memory which are storing the wallet and dev addresses. Nevertheless, is doesn't work quite right. I could upload the deobfuscated dll because the mentioned approach has a lot less code involved. Overall, I am not against a developer fee, but what this claymore is doing ( after debugging through his code) I don't like his statement of 1% dev which is actually far more - who else needs as one person around 500k $ per month for developing a cli program.... Just a hint for everybody : debug by x64odg with SycllaHide. You can search for the address in x64 by "ethdcrminer64:base + 0x9B29D". Kind Regards

**Comment by br1egel** [comment]
P.S: https://github.com/ethereum-mining/ethminer has a new release.
for r9 390 till rx580 one can use parameters: --cl-kernel 1 --cl-local-work 256 --cl-global-work 18432 --cl-parallel-hash 2
vega a running quite nice around --cl-kernel 1 --cl-local-work 128 --cl-global-work 18432 --cl-parallel-hash 4
&
--cl-kernel 1 --cl-local-work 256 --cl-global-work 65536 --cl-parallel-hash 4
global work size 65536 could be too big. Just wanted to mention ethminer, unless the dual mining and the gui is needed ...

**Comment by rzrwolf** [comment]
I had 24 hrs run with nofee on claymore 11 and it seems his protection works, as on 146mhs rig i got average of 136mhs on poolside.

Used as nodevfee.exe ethdcrminer64.exe and so on..

**Comment by rzrwolf** [comment]
[nodevfeeLog.txt](https://github.com/Demion/nodevfee/files/1726860/nodevfeeLog.txt)

Here is nodevfeelog.txt i got so far.

**Comment by br1egel** [comment]
same on my side, seems that hashrate is decreased. I have attached the dll ([ClaymoreLib.zip](https://github.com/Demion/nodevfee/files/1727083/ClaymoreLib.zip)) that had worked for v10 (be aware that I changed several int to work with v11). In  x64 dbg two wallet addr's
 (mine and claymores) are on the stack but referenced this time not in the same memory map of the program itself. I assume that he maybe delegates the saves to another program ? (just wild guesses - offsets could be 789024d or 722416d)

**Comment by Demion** [comment]
Maybe your devfee is mining to different pool, you can try pool redirection `nodevfeePools.txt`
I see diffrent connections in log:
```
sin_addr = 149.202.57.197 sin_port = 9999
sin_addr = 149.202.57.197 sin_port = 9999
sin_addr = 79.137.82.104 sin_port = 9999 
sin_addr = 79.137.82.104 sin_port = 9999
sin_addr = 149.202.57.197 sin_port = 9999
sin_addr = 79.137.82.104 sin_port = 9999
sin_addr = 164.132.108.171 sin_port = 9999
sin_addr = 164.132.108.171 sin_port = 9999
sin_addr = 185.71.66.38 sin_port = 9999
sin_addr = 79.137.82.70 sin_port = 9999
sin_addr = 92.222.180.118 sin_port = 9999
sin_addr = 92.222.72.197 sin_port = 9999
sin_addr = 213.32.74.230 sin_port = 9999
sin_addr = 149.202.57.197 sin_port = 9999
```
Maybe it is just mirrors of same pool (same host name). 
To be sure need to check bat / settings, `nodevfeePools.txt` settings (if used).
Claymore might have some hidden protection algorithms, I am no much expert in reverse engineering, for now I am still on 9.7 version.

**Comment by br1egel** [comment]
I think you did not understand me correctly : I am talking about the registers he passes to the function eth_submit work. That means, that it doesnt really matter what addresses he uses for his devfee, you just replace that memory content with another wallet address. I hoped that someone with a bit more reverse engineering skills could colaborate with me to get the right location. Nonetheless, I am back to ethminer - high avg hashrate with less stale share.

**Comment by Demion** [comment]
I understand what you mean, but my approach is similar in some way, have you looked at code?
It does hook (intercept) winsock2 (ws2_32.dll) win32 api and in send function replace their devfee wallet to your wallet. 
What makes you think if you patch that wallet in some other memory buffer or registry it will not get detected. Anyway good luck with that, feel free to share results here, would be intersting to see.

By the way if there is check for buffer integrity we can try use copy of buffer and leave original buffer unmodified or after packet sent just replace wallet back. I can create small test version, if anyone interested to give it a shot. Just remember you need to test for quite time (12-24 hours the least in my opinion), otherwise there a lot of variables that can affect actual hashrate.

**Comment by kirillp** [comment]
Я тоже самое делаю, копию буфера в своей ветке, я добавляю worker если его нет  и заменяю точки в воркере т.к мой пул иначе некорректно работает.

I do the same in my branch, I make a copy of buffer in case I need to add worker (if not defined). Also I replace '.' in worker names.  This is required on my pool (etherdig.net)

**Comment by br1egel** [comment]
alright, will look after that new code and try it. I found another approach, because I wanted to do the hack more at the origin instead of doing 'it' almost at the end, when alternating the submit. https://github.com/d3z00r/Claymore-11.0-No-DevFee-DevFee-Removed . That guy seems to have extracted a libeth.dll. Looks like claymore used virtualproctect to save his memory against access from outside. been testing this, hopefully the devfee function will never be called anymore -> that would be the best of all case

**Comment by Narratheon** [comment]
Please share your experience! Sounds pretty neat.

**Comment by Demion** [comment]
I would not advise using d3z00r version.
1. EthDcrMiner64.exe is unmodfied.
2. Only file seems to be modified is libcurl.dll (which is C++ http library).
3. According to hex editor and PE tools it is has IAT (import address table) modified so that it load (inject) libeth.dll
4. I havent reversed libeth.dll in details, but I believe it is 3rd party dll, not extracted from somewhere cause EthDcrMiner64.exe is unmodified and libcurl.dll has just libeth.dll inserted in IAT at end.
5. Most important in hex I found wallet **0x0b374e2f03dDe62aBdC7D28a9efa2081a93974aA**
https://ethermine.org/miners/0b374e2f03dDe62aBdC7D28a9efa2081a93974aA

So I believe it can be mallicious, I remove link for now. If anyone want to test then search for author name.

**Comment by br1egel** [comment]
well could be, but all I saw was that there was a 0x21000 address offset to the original ethdcrminer.exe and that in libeth.dll they were 2 methods only exported. the start method just returns zero. anyway, back to debugging :D    I liked the idea of dual mining keccak algos... I will try your new approach.

**Comment by Nickduino** [comment]
I didn't understand more than 10% of what I read here but I wish you guys all the best :D

**Comment by hardworker01** [comment]
I didn't understand much either but I will test NoDevFee_v0.2.6a and report back

**Comment by dugyitla** [comment]
i often get eth authorization failed whenever i start/restart claymore using 0.2.6a

ill observe for 24hrs. those who are testing, please report back as well. thanks

**Comment by dugyitla** [comment]
i hope you guys can make nodevfee efficient for 10.3 above. cant really use pre 10.3 because adrenalin is not supported with pre 10.3, thus cant oc, control fan, etc. and i have stability issues pre 10.3

**Comment by kirillp** [comment]
it is very efficient for me for 10.6

**Comment by br1egel** [comment]
Hi, seems to work. much lower stale rate in contrast to original claymore version 11. seems like claymore is still submitting the same share for his own devfee....
As hardworker01 already mentioned I sometimes get a json with unkown response (mining.notify etc.) After some restarts of the miner it works normally.

**Comment by br1egel** [comment]
Don't have the time now, but could be interesting to compare each binary versions ([snowman](https://github.com/yegord/snowman)). Maybe one gets a better idea of what claymore changes from exe to exe and isolate the devfee and/or dual mining part (keccak, blake2s).

**Comment by whitegoblin420** [comment]
Just wanted to drop in and leave my experiences for everybody. First off, thank you very much for your time working on this Demion, I really appreciate your open approach here. I've been testing your various releases for the last few weeks across some of my equipment including using the different versions of Claymore's software.

What I've found is the previous versions (v0.2.5b and down) produced very unstable hash rates inside 10.5, 10.6, & 11.0. (it's important to note: i've only ran tests so far while dual mining pascal lite & verge) To clarify this for others, what I mean by this statement, is my actual hash rates no matter of equipment, would rarely if ever match my reported hash rates. They would operate quite a bit below average, noticeably beyond the normal ups and downs expected throughout the process.

I always noticed the highest hash rates right after starting the program, but it would always die off quickly. You could restart the program to try and always keep the first few normal looking peaks/vally ratios but whenever you let it just work away you'd come back to having under performance issues. Almost as if, a delayed security check somewhere along the line kicked in & down with the hash rates it went after it noticed something afoul.

Though with the v0.2.6a build you posted for us to try utilizing buffer copy instead of modification everything is working at their reported hash rates or damn near it, hell sometimes over it. (aka: much closer to normal operation) I keep tweaking things myself which is not giving me the long term graphs I'd like to take screenshots of for reporting back here. Though once I get a decent screenie or two captured where they've had time to just run on their own for long(er) periods of time I'll return here to post about it.

This leads me to believe you're clearly onto something with this new technique. I wish I could help more but I'm not a skilled programmer. I'm benching your versions across my test rigs including; 13x 1080Ti FTW3 Hybrids, 4x MSI Wave Vega 64s, a Waterforce 1080, & a Titan Xp with it's own custom EKWB loop. They all regarding of manufacturer, driver version, clocks, currency being mined, or basically any setting universally like the v0.2.6a version better reporting more stable hash rates. I'm currently testing on Claymore's newest release (11.0), it would be great if other's could let us know their results as well.

These machines are running in Windows 10 Pro, latest builds released off the stable branch. They might throw a single rejected share a week and are generally rock solid. I notice the only times they tend to reject is when I'm personally testing on the machines as I'm taking resources away, turning monitors on and off, changing clocks, etc. When left alone they're good to go, this means they never throw the restart flags, which means I haven't had a situation come up where I could even test the restart.bat functionality. 

I do have it in there though as per your instructions file. So others can follow along I'm also including the nodevfeePools file, though I'm not using the nodevfeeWallet file. I'm on ethermine.org as my primary pool. I always delete the older preexisting log files before starting the miner again, though I have no reason to believe this does anything at all. (more just ocd about keeping a clean house so to speak) I've also, once again for no reason at all other then cheeky superstition, changed the name of the nodevfee.exe to hashishalldayallnight.exe and call it in the .bat file instead.

Not to go on a bicycle ride here but my thinking went back to cheating / hacking early MMORPG launches. In a couple Korean titles the botting and exploit software needed to be launched with completely randomized names and other identifying marks. There was a couple great bans that happen simply because they scanned for the same process names, process guards, etc.

Given there's been multiple versions of Claymore's releases since this has been around, I assumed at some point he would integrate some basic checks, at the very least, to see if nodevfee.exe or say even nodevfeeDll.dll where present in the miners launch location. Something to that overall effect anyways, I'd actually think he would integrate a system much more complicated then that, though once again, I'm not really a programmer.

I don't know if any of this information helps, I sure hope it does though. Good luck to everybody out there and of course, I'll be around watching! /shaka

**Comment by Demion** [comment]
Thanks for testing, great report.
I created small test update so that you can change not only **exe** name but also **dll** name. 
Names of exe and dll can be anything but must match though. 
For instance you can rename it to `RazerHook64.exe` and `RazerHook64.dll` (name is used by common Razer hook dll to monitor your key press stastics).
You can also try external injection method so that you use your original bat (without nodevfee), it might be less detectable as well.
> Create `nodevfeeInject.txt` with your miner file name inside; run `nodevfee.exe` without parameters; run your miner as usual (without `nodevfee.exe` before miner). Note: `nodevfee.exe` should keep running; `nodevfee.exe` `nodevfeeDll.dll` and all config files should be in same directory as your miner.

Other stealth techniques can be less trivial and less stable like - change code (dll) inject method, change hook (interecept) method to more advanced.

[NoDevFee_v0.2.6.1a_x64.zip](https://github.com/Demion/nodevfee/files/1739013/NoDevFee_v0.2.6.1a_x64.zip)

**Comment by dugyitla** [comment]
@whitegoblin420 mine results otherwise.

im gonna test it further.

**Comment by dugyitla** [comment]
@Demion

regarding stealth methods. I noticed that the original files were:
nodevfee.exe
nodevfeeDll.dll

thus, if i rename it to "example", should it be like this:
example.exe
exampleDll.exe

or:
example.exe
exampleDll.exe

which is which?

**Comment by dugyitla** [comment]
i mean:
example.exe
example.dll

why can't I edit post here?

**Comment by dugyitla** [comment]
and should I edit config files as well such as:

exampleInject.txt
exampleLog.txt
examplePools.txt

I want to try stealth methods, it might work, but I want to confirm first before trying, because a little downtime is not worth it.

thanks.

**Comment by Demion** [comment]
`example.exe` `example.dll`
You should see message in miner console right after start `NoDevFee v0.2.6.1a` to make sure it is injected and working ok.
Config file rename are not support right now, I will add in next update.

**Comment by dugyitla** [comment]
that answers my question well. thanks.

i see that i can edit post in pc mode. sorry for spamming your issue section.

thanks. testing it right now.

btw, i often see this error:
ETH: Authorization failed
: {"id":2,"result":false,"error":[20,"Notsupported",null]}

what does it mean? i see that using your nodevfee, but with my other rig, it doesnt show up. i dont know whats wrong.

**Comment by Demion** [comment]
Need full miner log file and `nodevfeeLog.txt` file to see whats going on.
I can guess it is known issue using pool connection redirection (`nodevfeePools.txt`).
DevFee uses different stratum protocols (`-ESM 0` / `-ESM 1`) from time to time, but your pool only supports one of them (only pool supports both at same time, that I know, is ethermine.org).
Only "solution" so far is set `-retrydelay 1` for faster reconnect and ignore this. 
Still need logs to check if that is the case.

**Comment by dugyitla** [comment]
i was on -retrydelay 1, so its reconnecting promptly. yes, im using nodevfeepools.txt for foolproofing. to make sure i still get the devfee incase devfee uses other pools.

so you mean i can ignore that?

thanks

**Comment by Demion** [comment]
As I was saying, need full miner log file and `nodevfeeLog.txt` file to be clear.

I created spoof version which replaces all devfee text strings for config files, error message boxes and even console messages.
Format will be - `example.exe` `example.dll` `exampleLog.txt` `exampleWallet.txt` `examplePools.txt` `exampleInject.txt`.

_Note: this is only test version may be unstable, may have critical bugs (crashes).
Honestly I think this wont do much, cause this sort of protection would be too trivial, but still as an option, shouldnt hurt._

[NoDevFee_v0.2.6.2a_x64.zip](https://github.com/Demion/nodevfee/files/1739338/NoDevFee_v0.2.6.2a_x64.zip)

**Comment by dugyitla** [comment]
woah that was fast. im currently testing the previous one. ill test the latest after 24hrs.

thanks

**Comment by hardworker01** [comment]
Currently dual mining eth+xvg with v.0.2.6.1a. Everything working great but have a question. Do I need another nodevWallet file for xvg wallet?

**Comment by dugyitla** [comment]
doing great. my 2hour-sma is leaning with my client's reported hashrate. now lets just hope claymore is not lurking here. lol. it just take another update and then boom, it will be patched again.

**Comment by h1dd3nsense** [comment]
Hi guys.
I'm just curious... Do you really think that the guy who makes $1M+ per month can't make a code that detects such kiddy's tricks? Ok... May be... At least until you share this code worldwide.
@Demion, what's the purpose of sharing this?
@br1egel, drop me a message to gmail, plz.

**Comment by Demion** [comment]
What is the purpose of writing such messages here? Please keep on topic.

**Comment by br1egel** [comment]
@h1dd3nsense 
[see](https://www.reddit.com/r/EtherMining/comments/6h25b8/is_the_claymore_miner_an_nsa_skunkworks_project/) post of PilsY86.
2.: claymore is busy making crypto to real money( try to cash out 1M$ per month w/o federal notice).

**Comment by dugyitla** [comment]
something fishy here.

anyway, @Demion, I now support you closing the source code. I know you are trustworthy.

**Comment by Demion** [comment]
What are you talking about? Closing source? 2.6 update? Source will be uploaded after tests, there is nothing to hide / to show, this alpha versions just for test, after they prove somewhat stable I will clean code and upload as usual with pervious releases.

**Comment by dugyitla** [comment]
this dev is upfront. i hope you continue developing this app. thanks.

**Comment by Demion** [comment]
I have no time and intention and maybe skills to devote beating claymore (or any other miner) protection techniques (if there are any). 
I created simple network packet interecption patch tool a while ago for private use and then just decided to share with community as it proved to work pretty well (at least on 9.7, which I am still using). 
You are doing right thing though, respect developer and pay fees (no sarcasm). I dont think his income will decrease much if dozen people use nodevfee, look at his dev fee wallets. I also think donations should be by free will, not forced fees and programs open sourced for futher development & innovation.
Then it grew a bit with pool redirection and some other fixes and configs. By the way it has a lot of config files, which is ugly, because it was developed with idea of being simple as possible and backwards compatible (you can still run it without a single config txt file). 
For future there is no much can do, only can experiment with things I mentioned like playing with different inject and code hook methods.
Ofcourse it will be always open source and if someone manages to find way to "hack" future versions I will merge / release this update here.
_At that point can already create own miner (fork ethminer) with blackjack and colorful console messages, with remote web interface and neat config features, only thing it will lack is dual mining._

**Comment by minahan24** [comment]
Hi Demion,

First of all, thank you for your work. Does it work using MPH pool. I tried version 2.6.1a and rename the exe and dll. And then added my wallet in the nodevfeewallet.txt, I have the nodevfeepools.txt and I moved the address of the miningpoolhub to the first one. I also have the nodevfeeinject.txt with Ethdcrminer64.exe on it. But I'm getting this error. Looks like it is redirecting but using a different wallet. If you are familiar with MPH the wallet uses your username+rig name.

DevFee: ETH: Stratum - connecting to 'us1.ethpool.org' <149.56.26.222> port 3333
NoDevFee: connect -> us-east.ethash-hub.miningpoolhub.com:20535
NoDevFee: eth_login[1] -> 0xdE088812A9c5005b0dC8447B37193c9e8b67a1fF

Received unknown response: {"id":null,"method":"mining.notify","params":["c2a6","0x8144f1974fd0267cf284eec78d725a84ff0d4a999e3c5da75a1695be57442885","0x4e99a30e99712c8c6e292fe7ba6b27a37c7ced12e2ec7862f31fb676724cb404","0x00000000ffb34c02420e9948eacd78cf33b059a88ade1ff0614f7f3c303cf3a7",true]}

Here is my log file.
[nodevfeeLog.txt](https://github.com/Demion/nodevfee/files/1741492/nodevfeeLog.txt)

**Comment by h1dd3nsense** [comment]
>>What is the purpose of writing such messages here?
The purpose is pretty clear. You are not only one who uses such techniques. However you are the one who is making impossible to use this in the future.

**Comment by br1egel** [comment]
that's the tradeoff : trust a program, because you can see the source code or trust some programmer, which says 'I obfuscated the program to save you, that it will still work'. I would choose the first one, despite having to crack that exe from version to version.

**Comment by h1dd3nsense** [comment]
@br1egel that's fine in an "ordinal" case. But in this case the "tradeoff" leads to uselessness of the software.

**Comment by whitegoblin420** [comment]
Reporting back in after testing things longer. Sadly coming back to tell you all I was wrong and the rates I experienced originally would not hold long term. I've taken two screenshots to show people what I'm talking about. The first is everything setup exact from my previous post (though only across three test machines now) while running v0.2.5b, the second is running v0.2.6a.

 These are both while dual mining blake2s across 4x1080Ti, 4xVega64, 1xTitanXp, 1x1080. I've left the other half of the cards out of these runs as they're sitting on more profitable coins right now to make up for the downed hash rate during these testing times. Kinda trying to balance the scales a bit so to speak on the weekly returns while still providing a helping hand here.

**Test 1:** [NoDevFee v0.2.5b - Claymore V11 Dual Mining blake2s - ETH Results](https://i.imgur.com/p0pvC1B.png)
**Test 2:** [NoDevFee v0.2.6a - Claymore V11 Dual Mining blake2s - ETH Results](https://i.imgur.com/FFFBfqy.png)

 Now that I've had a second to get caught up and read what everybody has been talking about I've downloaded a copy of v0.2.6.2a, skipping the previous. I clearly read your warning about it being a test version only and prone to crashes / instability. I'll let you know what happens! I've changed all the names over to your example convention so that later screenshots will be easier to follow along for others.

 I wanted to comment that I never get the auth failures folks are discussing. Though I think this is because I'm using ethermine.org an they support both stratum protocols (-ESM 0 / -ESM 1) as pointed out by Demion. So for this round of testing I'm using all options; example.exe, example.dll, exampleLog.txt, exampleWallet.txt, examplePools.txt, & exampleInject.txt.

 I'm as suggested starting v0.2.6.2a up first and letting it's window just hang. Then I fire up Claymore's V11 as I would normally, not calling example.exe beforehand or anything of the such. Right away I clearly see _"EthDcrMiner64.exe" [2732] -> "example.dll"_ pop up and we're off to the races. Everything appears to be running normal, hash rates are high, hopes are even higher! I'll be sure to report back in after another 12-15 hours or so of running.

 On a more personal note I'd like to say thanks one more time to Demion. After reading other's posts on the net about trying to out thwart Claymore I think people tend to get the wrong impression by the few people who do lend a hand to the situation. I mean outside of the kicks of doing such a thing & the extra financial motivations it provides, I could only imagine somebody would invest an excessive amount of time into this project to measure if Claymore himself was being honest in his releases.

 There's a popular opinion floating around the net that he's got his hands more then balls deep in the cookie jar depending on how you configure his misc. mining programs. Though if chasing him around in your free time trying to make a name off all that isn't your thing and you were merely sharing this wonderful foundation for all of us to grow from then that's pretty bad ass too. I know I can think of alot of things I'd rather be doing then hacking away at a mining program and there's no reason you shouldn't get to share those same experiences / thoughts.

Thank you for everything you've done so far and for anything you do in the future. /cheers

**Comment by br1egel** [comment]
@whitegoblin420 
I see that you have 4 vega64. what are your settings for eth mining (gpu core and gpu mem) ?
for cryptonight I always get the reported hashrate but for dagger hash its always under the reported hashrate. unfortuntelay hwinfo64 can't tell if there are memory errors.

**Comment by whitegoblin420** [comment]
Alright it's been roughly 17 hours of testing the new version. Everything has been left the exact same as my previous post detailed. I believe we now have a winning concept on our hands as I'm seeing stable hash rates across all the machines! (or atleast expected hash rates) The main counter argument one could make is that I'm still seeing subpar hash rates on the Vega 64 rig. Though if you read the Claymore release notes it specifically says _"KNOWN ISSUES: AMD cards: on some cards you can notice non-constant mining speed in dual mode, sometimes speed becomes a bit slower. This issue was mostly fixed in recent versions, but not completely."_ So keeping that in mind I firmly believe Demion has succeeded with v0.2.6.2a inside Claymore V11.0. 

**Test 3:** [NoDevFee v0.2.6.2a - Claymore V11 Dual Mining blake2s - ETH Results](https://i.imgur.com/NJkl0r1.png)

As always, more testing can & will be done, but that's pretty solid compared to the previous tests. As you can see the biggest rig is running on average 10 MH/s below it's reported rate though that's across seven cards including four vega 64s. I feel like the drop Claymore discusses in his release logs is to blame here for that difference, not a security system flag or failure like we saw previously. The 1080Ti machine (a gaming machine I turned on to test things with) is running an average hash rate of 1.2 MH/s below it's reported rate while the Titan Xp + 1080 machine is also interesting enough running on average 1.2 MH/s below it's reported rate as well. 

These differences are then being made up with the dev cycle's "default" worker if I'm understanding everything correctly. So there you have it, stable & correct ETH results comparable to running vanilla. The rest of the swings you see on the graphs I'm speculating are from incorrect mining parameters, overclocking, or even environmental issues on my part that I've yet to dial in. (ex: -ethi, -eres, -dcri, etc. settings) I'd be open to anyone else's interpretation of the results but I'm seeing the light here compared to the previous tests.

 This should give others a proper starting grounds for running the same test as I have laid out here. With additional results we should be able to more amply dial into what's working and what's not. Though this was clearly a very positive run on my end and I'm excited for what it means moving forward. If anyone out there reading this would like to follow up and post some of their own results I know it'd be greatly appreciated by the community.

 Also before finishing up, @br1egel I'm running 1500 core / 1100 memory. Custom power play tables I created using the popular excel tool then once undervolted, I had to up overall power availability to maintain stability during the dual mine. They never budge off those clocks an don't get to hot but keep in mind I'm on water an running them at 90% pump/fan speed as well. Looking at the stats screen they're averaging 43.444 / 955.92 on the current dual mine of blake2s.

 Though just for the notes of anyone who happens to be reading this, from my personal experiences, dual mining pascal requires significantly more power and there for heat output to maintain stable rates compared to blake2s. So make sure you take that into account before running prolonged tests on your hardware if you happen to be exploring dual mining. /crescent

**Comment by jsm0** [comment]
I've been running 2.6.2a for just over 24 hours. I would like to confirm that it is working. I'm using claymore v11, dual mining pascal (mining verge caused hashrate to jump around too much). On my 13 570 rig, my 24 hour calculated average hashrate is higher than my reported hashrate (402mh/s reported, 405 mh/s 24hr). Much appreciate all you've done Demion. Will continue to run.

**Comment by PeshBG** [comment]
i get loadlibraryW error#0 with the 2.6.1 or 2 version

**Comment by Demion** [comment]
@PeshBG dll name must match with exe name, rename `nodevfeeDll.dll` to `nodevfee.dll` or any other name `example.exe` `example.dll`

**Comment by PeshBG** [comment]
sleepy me. i just extracted your zip :) your dll is namedll.dll :D

**Comment by Demion** [comment]
_Cleaned topic, too much offtopic and flood hard to read._

**Comment by minahan24** [comment]
@Demion I'm using -esm 2 for MiningPoolHub, like you said, it's not supported yet. Too bad, but I will patiently wait and hopefully you support it in the future. I like using MiningPoolHub because of the AutoExchange feature. But if push comes to shove, I will just move to Ethermine pool.

Again, thanks for your hard work.

**Comment by hardworker01** [comment]
@Demion how can we contact you other than here?

**Comment by h1dd3nsense** [comment]
@hardworker01, as an option, you can contact me ;) - the same nick @gmail.com

**Comment by Demion** [comment]
Made contact email available at https://github.com/Demion

**Comment by P-D-A** [comment]
@Demion 
Приветствую 
Я по-прежнему использую PM и NoDevFee 0.2.4.
На последней версии NoDevFee_v0.2.5b_x64.zip
значительно проседает средняя скорость примерно на 15-20MH
Поэтому я использую предыдущую версию NoDevFee_v0.2.4b_x64.zip
И получаю действительную скорость а иногда даже выше
Я забил на на баг подмены адреса пула
http://ibb.co/hvNPoH
http://ibb.co/jst8Fx

**Comment by Demion** [comment]
Был изменен только OnConnect, выполняется он только при новом подключении, поэтому проседание скорости очень мало вероятно, возможно просто "плохая удача" или что-то с пулом, совпадение. Но и 0.2.4b должна работать в большинстве случаев.
Можете посмотреть какие изменения были в этом обновлении - https://github.com/Demion/nodevfee/commit/3cd6ee407b24d406f7ca24960dc4fb495ed0acec

**Comment by P-D-A** [comment]
Да я видел изменения 
И пробовал менять версии несколько раз 
Как только я повышаю версию у меня уменьшается среднее количество шар
При понижении версии количество шар увеличивается

**Comment by krioTT** [comment]
Hi Demion,

All 3  v0.2.6.2xx versions are crashing the computer after approx 2 hours,
[with internal driver error and need hard reboot]
happens x4 times in a row when using any of those versions.
OS: Win7 64bit

If i don't use the the tool or use any older version 0.2.5b or below,
all works fine and no any issues for many hours. [over 150 hours + more]

On other 3 rigs with Win10 64bit v0.2.6.2xx versions are working fine,
so i believe it has something to do with OS

**Comment by rzrwolf** [comment]
I am still getting lower average hashrate than without nofee.. Used last 0.2.6.2x versions with rename, nodevfee pools. I am using bat with nodefvee.exe ethdcr.exe style launch.. 

Нормально работает только с 10.0 версией.

**Comment by mayner88** [comment]
Does not want to shell 0.2.6.3a cleymore v 11 on the fork pool -allcoins -1 redirects to another pool Well, the ETZ wallet is not ETH

**Comment by mayner88** [comment]
It can not be broken in any way so that he does not cause devree?

**Comment by Nickduino** [comment]
So, I extracted both files from _NoDevFee_v0.2.6.2a_x64.zip_, renamed them _this.exe_ and _this.dll_.
Created _thiswallet.txt_ with my wallet (same as my main wallet) address, _nodevfeeinject.txt_ with Claymore's file name (by the way, you could say in the readme it needs the ".exe", so "_EthDcrMiner64.exe_". Or have your program add it itself if it's not in the text file).
...and it seems to work.

So, should I bother with the pool text file or can I let it be and expect the devfee to be mined on my nanopool account?

**Comment by P-D-A** [comment]
@Demion 
Думаю актуально 
http://ibb.co/g1WSAx

**Comment by SiegHeil** [comment]
Предлагаю отказаться от DLL с хуками, а взять за основу WinDivert и орудовать на более низком уровне

**Comment by Demion** [comment]
Хочу попробовать LSP перехват, правда его не правильная установка может "убить" систему. WinDivert какую технологию использует? Выглядит интересно.

**Comment by P-D-A** [comment]
По-моему уже даже есть 
https://github.com/GTANAdam/NoDevFee
Надо собрать и проверить

**Comment by Demion** [comment]
**WinDivert** test version (using _Windows Filtering Platform_ driver), no connection (pool) redirection yet, no logs.
No need to inject, no need any config files, can be located at any folder, just run `divert.exe`. 
If you want to specify wallet directly run `divert.exe 0xcb4effdeb46479caa0fef5f5e3569e4852f753a2`

Thanks for idea **SiegHeil** for idea and **basil00** for core library.

[divert.zip](https://github.com/Demion/nodevfee/files/1754353/divert.zip)

**Comment by dugyitla** [comment]
currently testing, how will i know its working?

i just run it, and then run my unmodified start.bat

i put it in separate folder.

i can no longer see any difference in claymore's logs.

is that how you supposed to trick claymore? to do not alter any from it? so only in pool can i confirm that its working?

i just run divert.exe without adding my wallet, will it automatically detect my wallet?

**Comment by Demion** [comment]
It should detect your wallet same as regular nodevfee, unless you are using altcoin / fork / dual mining (in that case devfee might be first miner connection / packet and wallet deduced wrong). 
You should see similar nodevfee messages, but in divert.exe console, miner will be untouched.

**Comment by dugyitla** [comment]
aw, i was dual mining. so i rerun everything, here is what i did:

1. create divert.bat, inside it is:
divert.exe walletaddress
2. run divert.bat
3. run claymore.bat(unmodified)

but i can see no difference now, its like untouched claymore console without those nodevfee logins. is that normal?

**Comment by Demion** [comment]
Dual mining should not be a problem, I think, but still you can set wallet directly, to be sure.
As I said, messages should appear in `divert.exe` console, not miner. Forgot to mention, that `divert.exe` should obviously keep running all time.
https://i.imgur.com/WAmR7Y0.png

**Comment by br1egel** [comment]
thanks for finding this. I guess, everything will be fine until ssl connection is mandatory for eth ?

**Comment by dugyitla** [comment]
it seems not working for me:

https://i.imgur.com/0CbK8JY.jpg

**Comment by Demion** [comment]
@br1egel I have no idea, I cant even confirm that this version is 100% undetectable and hashrate will be top notch. I hope testers will tell more. But you are thinking in right direction.

@dugyitla 
It is built on top of 3rd party library **WinDivert**, so I cant tell for sure. Working for me, but you can try disable antivirus (if any), windows defender, user account control, and first of all try Run As Administrator.
> ERROR_ACCESS_DENIED | 5 | The calling application does not have Administrator privileges.

According to https://reqrypt.org/windivert-faq.html

**Comment by dugyitla** [comment]
everything you said has been set properly.
no antivirus, defender whatsoever, uac set to 0, running the divert.bat closes the console immediately.

**Comment by Demion** [comment]
Whats Windows version exactly? Are you running As Administrator? 
I cant help more for now, lets see if this is common problem.
I may test in on virtual machine with your windows version to try reproduce error.

**Comment by dugyitla** [comment]
im on windows 10 1709

tried running the divert.exe as administrator, the claymore cannot connect to pools.

**Comment by dugyitla** [comment]
do we not need installing any cert?

**Comment by Demion** [comment]
I reproduced problem on virtual machine both on Windows 8.1 and Windows 10 1709.
For some reason it does work on my main PC (Windows 8.1).
No working solution so far.

**Comment by dugyitla** [comment]
i see. well, my stupid guess is maybe it need some sort of whatever signing on a specific machine where the app will run.

**Comment by P-D-A** [comment]
@Demion 
На rig ещё не закидывал но на win7x64 тоже работает

**Comment by Demion** [comment]
Downgrade to **WinDivert Version 1.3.0** solved my problems on test virtual machines (both 8.1 and 10 (1709)).
[divert2.zip](https://github.com/Demion/nodevfee/files/1754641/divert2.zip)

**Comment by dugyitla** [comment]
why removed linked sir? i want test it.

**Comment by gubenena** [comment]
just a silly question, divert.exe should be started from a .bat file<> divert.exe 0x7.............. wallet address. 
Thank you for all the effort !

**Comment by dugyitla** [comment]
tried just now. its showing the eth submit login now. will observe longer.

in the future, in case we use fork coins, im still confused on how to specify specific wallet.

creating bat file divert.exe wallet

running the bat file as admin closes it immediately. however, setting divert.exe compatibility mode to run as admin, running the bat file opens 2 console, which is confusing if it will work correctly.

**Comment by Demion** [comment]
@gubenena Yes. Do you know another way? :smile: 

@dugyitla 
1. You can disable UAC completely (_modern Windows apps wont work_) `regedit.exe` -> `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Policies\System` -> `EnableLUA` = `0`.
2. You can set always run as administrator `Properties` -> `Compatibility` tab -> `Run This Program As An Administrator` check box.

2.2 update will prompt administrator automatically (_still need to allow_).
[divert2.2.zip](https://github.com/Demion/nodevfee/files/1754700/divert2.2.zip)

**Comment by dugyitla** [comment]
yes sir,  i did edit the properties of divert.exe to run as admin, so it will always run as admin even just double clicking it. LUA is 0 from the beginning. 

so here is the problem, i created a bat file to specify an eth wallet
divert.bat:
divert.exe 0xwalletblablabla

so, when i run divert.bat, it runs 2 console, is that fine?

one is the bat file console, and the other one is the divert.exe console

**Comment by P-D-A** [comment]
1 we use the full path to divert.exe
Example C:\divert2\divert.exe 0xcb4effdeb46479caa0fef5f5e3569e4852f753a2
2 create a shortcut to the bat file and already on the label we say to run from them admin
[Example_bat.zip](https://github.com/Demion/nodevfee/files/1754720/Example_bat.zip)
![2](https://user-images.githubusercontent.com/20209313/36633351-1a0da256-19a5-11e8-998c-b5af3480268d.JPG)

**Comment by Demion** [comment]
I think is still some issue with rights, because I am having same situation on virtual machines, but no problem on my main PC where UAC is disabled completely. I think cmd.exe has no Admin rights that is the problem. 
Anyway you can try `start divert.exe`.

**Comment by dugyitla** [comment]
yes thats my guess as well, in win 10, cmd has no admin rights as default.

anway, in shortcut link, adding parameters just as easy as pasting the wallet right?

**Comment by Demion** [comment]
@dugyitla Sounds good, you better test on practice how it works.

Added connection (devfee pool) redirection.

File name - `hosts.txt`.
Format - same as `nodevfeePools.txt`.
Location - should be in directory with `divert.exe`.

_Hopefully this is not big overhead for traffic filtering. This is very experimental / unstable version. Enjoy._

[divert3.zip](https://github.com/Demion/nodevfee/files/1755202/divert3.zip)

**Comment by SiegHeil** [comment]
Need rename the WinDivert driver for reliability ;-)
Ну а то мало ли чего

**Comment by P-D-A** [comment]
Предлагаю переехать https://github.com/Demion/nodevfee/issues/34
А то листать много

**Comment by minahan24** [comment]
@Demion 

Not sure if the divert works on MiningPoolHub. I remove my -allpools 1, I have my username.RIG after the divert.exe on my bat file. I also have the hosts.txt.

And this is my divert.exe console. I can see the redirection, but there's no eth_submitlogin

https://imgur.com/a/e3haz

**Comment by Demion** [comment]
@minahan24 I guess you are using `-ESM 2` which is not supported. Only supported protocols are `-ESM 0` and `-ESM 1` only with Ethereum wallet, usernames not support as well yet.

@SiegHeil Особо нет смысла переименовывать. В Description и Product Name все равно можно найти по ключевому слову WinDivert. Чтобы это "скрыть" нужно драйвер пересобирать тогда.

> Driver Name       : 0MIDbcs3
> Address           : 00000000 FC9D8000
> End Address       : 00000000 FC9E6000
> Size              : 0x0000e000
> Load Count        : 1
> Index             : 158
> File Type         : Network Driver
> Description       : WinDivert 1.3 https://reqrypt.org/windivert.html 1C5vZVSbizPeZ8ydTYhUfm4LA2cNwBfcYh
> Version           : 1.0.0.0
> Company           : Basil
> Product Name      : WinDivert 1.3 driver
> Modified Date     : 17.10.2017 15:02:25
> Created Date      : 25.02.2018 8:56:48
> Filename          : C:\Users\Admin\AppData\Local\Temp\0MIDbcs3
> File Attributes   : A
> Service Name      : 0MIDbcs3
> Service Display Name: 0MIDbcs3
> Digital Signature :

**Comment by SiegHeil** [comment]
@Demion Понятно, да и ведь правда логично.

> Only supported protocols are -ESM 0 and -ESM 1 only with Ethereum wallet, usernames not supported as well yet
А в чём проблема распарсить JSON (судя по логу там в пакетах именно данные в этом формате летят)? В этом случае и воркера можно добавить, да и подмену кошелька на юзернэйм. Также возможно будет попроще "ESM 3" найсхэшевский добавить для жаждущих :-)

**Comment by Demion** [comment]
@SiegHeil проблем нет, времени не было добавить и тестировать. 
Единственный вопрос есть ли в майнере devfee пулы с поддержкой ESM 2 / 3.
Если нет, то нужно будет весь devfee трафик переделывать под другой протокол, т.к. devfee будет использовать пулы с ESM 0 / 1 и слать пакеты в соответствующем протоколе.

**Comment by minahan24** [comment]
@Demion

Yes. I'm ysing ESM 2 for MiningPoolHub. Based on your previous comment. If I dont use the -allpools 1. I can confirmed that claymore have an account with MiningPoolHub and devfee is mining to that account using claymore.efee username. If you can make a version that works on ESM 2, I will gladly test it.

**Comment by SiegHeil** [comment]
@Demion Последний вариант с WinDivert как-то стабильнее работает: погонял сутки и судя по статистике хэшрейт принятый пулом всего на чуть-чуть меньше того, что передаёт майнер

**Comment by dugyitla** [comment]
@Demion

may I suggest you include in divert console that every eth submit login, it will show on how many times it did it.

for example:
eth submit login x1
eth submit login x2
eth submit login x3

and so on, so i can verify that it did not freeze or ran some problem. or it would be better to display what time it functioned. 

because im doubtful sometimes, in pool's report, "x" worker eventually submitted less share than average. so i end up restarting divert.exe to make sure.

thanks.

**Comment by Nickduino** [comment]
Same for nodevfee. It would be nice to see it...with a time stamp.

And, if you're bored, a countdown to the next devfee period :D

**Comment by dugyitla** [comment]
@Nickduino
great, thats what im trying to say.

i hope @Demion will consider this one.

**Comment by SiegHeil** [comment]
@Demion мне тут в голову пришла мысль: а не проще ли вместо **connect** делать подмену еще раньше, чтобы знать к какому адресу осуществляется попытка соединения (**gethostbyname**???). Просто заметил, что IP каждый раз запрашивает при очередном connect'е и временами он отличается от того, что был получен на этапе инициализации и соответственно редиректится.
Это также вероятно позволит назначать в качестве адреса пула любой иной для перенаправления "dev fee"

**Comment by eyeflip** [comment]
![Uploading Screenshot (36).png…]()

I keep getting this error. not sure how to fix. I tried the first and second divert files you posted, and I get this error. the third divert file disconnects the internet.  I also run the divert as admin and I set the hex to 0. any ideas on what I might have done wrong? 

thanks for all your help!!

**Comment by Demion** [comment]
@dugyitla Will add timestamp in future builds.

@eyeflip I dont see your screenshot reupload to imgur.com

@SiegHeil не очень понял суть вопроса. Подменять результат **gethostbyname** не получится т.к. там только адрес, а нужно совпадение по порту тоже (т.к. может майнится дуал коин на тот же пул, но порт другой). Новая API **getaddrinfo** используется ни везде. Можно использовать **gethostbyaddr**, но зачем?

Проблем с перенаправлением не наблюдал, **gethostbyname** получает список всех адресов по домену (я так понимаю там несколько зеркал может быть привязяны к домену Round robin DNS), и потом перебирает, просто они иногда в другом порядке подаются. 

В WinDivert перенаправляю так же, но список адресов получаю только один раз в начале, чтобы не замедлять работу.

**Comment by SiegHeil** [comment]
@Demion по поводу подмены DNS запросов: кажется встречал данный функционал в [goodbyedpi](https://github.com/ValdikSS/GoodbyeDPI/), который также использует WinDivert.

`не очень понял суть вопроса. Подменять результат gethostbyname не получится т.к. там только адрес, а нужно совпадение по порту тоже`
Забыл про порт, а так идея была проверять хост не по списку полученных IP, а именно по его имени. Про хуки пример привёл чтобы упростить понимание сути идеи (но не угадал).
А вообще я просто задумался над тем, как перенаправить devfee на другой пул, т.к. текущий используемый в качестве логина использует не кошелёк, а имя пользователя ;-(

**Comment by P-D-A** [comment]
Я использую обратную запись ip
Так более стабильно

**Comment by Demion** [comment]
Изменение результата запроса DNS тоже может не верно работать, если там запрос идет только по имени, без порта. Посмотрю исходники, спасибо.

Замена должна работать нормально, перенаправляется на первый (`0`) адрес, который выдает `gethostbyname` для основного пула. В будущих версиях, буду перенаправлять devfee на последний использованный адрес основного пула.

Так же надо будет добавить таблицу перенаправлений, сейчас может работать не верно, если, например, нужно перенаправлять больше 1 соединения одновременно.

1. Получить список всех адресов по имени. Сравнить текущий адрес со списком всех адресов. Заменить при совпадении.
2. Получить имя хоста по адресу. Сравнить результат с списком хостов. Заменить при совпадении.

Вроде тоже самое, только наоборот. Единственное, что перебор должен быть более быстрый (адресов больше, чем хостов, т.к. к каждому хосту несколько адресов), но не думаю что это должно так влиять на стабильность и скорость. 
В любом случае в новой версии планируется получать список адресов только при запуске (или лучше с определенной периодичностью), а не при каждом подключении. (тем более в WinDivert нужно менять адрес каждого пакета, а не только при подключении).

С добавлением json парсера будет имя devfee воркера, поддержка других протоколов и возможно изменение devfee трафика под нужный протокол. Так же можно добавить перенаправление devfee на отдельный пул.

@eyeflip Your log looks very weird. According to log you are using supported protocol (-ESM 0 eth_SubmitLogin). But socket value and buffer value in hex looks very weird. Are you sure you are not using some sort of antivirus / firewall software? Never seen such behaviour before, especially you saying it is working fine on other PC, as I understand.

**Comment by whitegoblin420** [comment]
Alright so I've been testing the Divert3 release for just shy of twenty-eight hours now. This has been done across four Vega 64's and four 1080Ti's while inside Windows 10 Pro Build 1709. They've not been touched since the test was launched. I created a shortcut to the start.bat file for launching with my wallet address. The entire Divert folder is isolated away from the Claymore folder (aka, it's not inside the claymore folder). I'm using the hosts.txt file & I'm still running the tests over on ethermine.org. I cleaned out the Claymore folder of any third party files that didn't come packaged with it's release and all previous log files beforehand.

[Test 4: NoDevFee Divert3 - Claymore V11 Dual Mining blake2s - 14+ Hour ETH Results](https://i.imgur.com/V3Mwm3p.png) _(starts at up spike)_ 
[Test 4: NoDevFee Divert3 - Claymore V11 Dual Mining blake2s - 27+ Hour ETH Results](https://i.imgur.com/bU7hXLo.png) _(runs entire graph)_

 I've not had any problems getting it to work as it's all very straight forward. I'm going to have to agree with @SiegHeil here, this release does appear more stable then the previous tests I've ran. As you can see I'm not experiencing any of the hash rate drops seen in tests one and two. With these results I'd consider this a win!

 Speaking for the test results themselves, I'm still closing the gap on reported hash rate vs. average hash rate. Same as my previous posts I feel the difference we're seeing here in the screenshots is probably more or less configuration error(s) on my part.  I don't feel it's on any behalf of NoDevFee's interaction or we would've seen the rather large hash rate drops from before. Though to be clear I have fine turned my launch parameters a little bit since my last tests where ran.

 To be honest I was reading the claymore release thread and people where getting quite a bit better secondary blake2s hash rates then I was seeing so I suspected I might have noobed it up. Using dcri 85 (vega) / 125 (1080ti) I'm able to get hash rates of 43.355 / 3685.545 on the Vega64 (@ 1500/1100) & 36.656 / 1527.223 on the 1080Ti (@ 2075/5622). Given previously I was getting a quarter of that blake2s rate across the Vegas that's a pretty decent improvement.

 I'm speculating I wouldn't see such large peak:valley swings in my ETH results if I dialed it back down a bit though I really don't know. I guess that'll be a test for another time & place as I'm starting to get away from the purpose of this post. Just wanted to give as much info as I could for the time being. Thanks again everybody, great to see all the progress being made! /420BIF

 **TL;DR** _Divert3 works in Claymore 11_

**Comment by kirillp** [comment]
Не нужно добавлять json  парсер. В моём форке добавляется варкер простой функцией. Возми код от туда или давай я сделаю пул реквест. Отлично добавляется и работает уже у меня давно.
https://etherdig.net/#/account/0xe5a42563432efe1774f87423570b48cd6181f24f

**Comment by dugyitla** [comment]
claymore is really focusing on his $$

he updated his app to support ssl. he might be asking pools soon to make ssl mandatory.

**Comment by Nickduino** [comment]
@Demion Can you clarify the divert vs nodevfee situation? Is there a reason to use one over the other?

**Comment by Demion** [comment]
**divert** version is based on **WinDivert** driver by **basil00** whish is using _Windows Filtering Platform_.
It is low level driver which intercepts and modifies network traffic. Should be less detectable and overall better approach as there is no direct interaction with miner (no dll injection, no winapi function hooks).
It is very alpha, test build, may be unstable.
If it proves to be solid, nodevfee will be most likely replaced by WinDivert implementation in further releases (v.0.3).

Claymore v9.7 seems to work fine with nodevfee, but starting from 11.0 or even 10.6 users report slowdown in actual hashrate after some time (I seen that too, but never tested enough). 

**tl;dr** Test and use whatever works best for you. Both version do work for me. If you are using version 10.6 or later and notice hashrate slowdown after time you may give new version a try, but keep in mind this is test build.

**Comment by Nickduino** [comment]
That clarifies it, then, thanks!

**Comment by Hackintoshihope** [comment]
When using WinDivert all I get are rejected shares... And the DevFee mining never stops. I sthis because I am on a wifi connection or is something else going on?

**Comment by SiegHeil** [comment]
@Demion 
>'Кстати только заметил 16 и 0 адрес одинаковые. hosts.txt правильно настроен? Или у разных хостов один адрес и порт? Странно, там же не HTTP протокол.'

Использовал файлик с хостами от предыдущей версии, там есть повторяющиеся хосты с разными портами, потому подозреваю и совпали... а может это некая балансировка

**Comment by ioMinx** [comment]
Thanks @Demion . Testing divert3 so far so good. I will give more detail report if it's passed 24 hours..
Btw, can I use the following format after the divert3.exe <wallet add>.<worker name>.<email> ?

**Comment by Hackintoshihope** [comment]
![rejectedethshares](https://user-images.githubusercontent.com/36898635/36767110-1167ce7e-1c07-11e8-8796-4d68eb5e5426.png)

Here is what I keep getting everytime the devfee pops

![hosts](https://user-images.githubusercontent.com/36898635/36767150-40a159a8-1c07-11e8-8a5f-41986a4a65fb.png)

Here is the host I want it to connect to but it doesnt seem to do that.

**Comment by ioMinx** [comment]
![screen shot 2018-02-28 at 10 02 28 am](https://user-images.githubusercontent.com/36903456/36767782-8cc27684-1c6e-11e8-92bc-f577ad0801e6.png)
This is mine.. no rejected shares

**Comment by Hackintoshihope** [comment]
Could you share your batch file to start divert and your host file you are using? Maybe I’m doing something wrong... thanks.

**Comment by ioMinx** [comment]
its a standard divert3.exe <wallet adds>

i can see that you are dual mining, if you read above post, dual mining isnt supported yet

**Comment by Demion** [comment]
@Hackintoshihope `hosts.txt` is wrong. It can never be 1 line. Read readme again (`nodevfeePools.txt` format). Dual mining should be supported.

@SiegHeil судя по скриншоту в консоли хост и порт одинаковые же 16 и 0.

**Comment by Hackintoshihope** [comment]
@Demion trying with updated format.

**Comment by goodbrat** [comment]
![image](https://user-images.githubusercontent.com/36914985/36776757-f0937b12-1ca1-11e8-8597-ed2b3bce673d.png)

What's wrong with this version of 2.6? Error#0

**Comment by PeshBG** [comment]
rename the files to have the same name 1.exe 1.dll for example

**Comment by Demion** [comment]
Are you using **0.2.6b**? dll and exe name should match, as **PeshBG** said. 
Also should be in same folder as miner. Check for any antivirus software, UAC, Windows Defender. 
Try set your miner at `C:\` сyrillic or hieroglyphs may be not supported, use latin (english) paths or root (`C:\`) to be sure.

### Create new issue this is not global topic to discuss any problems. I will close this soon.

**Comment by P-D-A** [comment]
Я что то неправильно сделал ?
![image](https://user-images.githubusercontent.com/20209313/36779172-322c8356-1c80-11e8-876c-24caaab7333d.png)
![image](https://user-images.githubusercontent.com/20209313/36779381-dfcbabae-1c80-11e8-8df9-0806917adcbb.png)

**Comment by Demion** [comment]
Явно что-то не так. Скидывайте сразу config.txt, иначе угадывать приходится.

**Comment by P-D-A** [comment]
[config.txt](https://github.com/Demion/nodevfee/files/1766425/config.txt)

**Comment by Demion** [comment]
В конфиге
```
eth-eu1.nanopool.org 9999
eth-eu2.nanopool.org 9999
```
А на сриншоте
`29.ip-5-196-13.eu 9999`

**Comment by P-D-A** [comment]
@Demion

![image](https://user-images.githubusercontent.com/20209313/36779851-65c72020-1c82-11e8-8e53-e52881dbe478.png)
основной пул в PhoenixMiner и в divert3.1 должны соответствовать ?

**Comment by Demion** [comment]
Main Pools - это список ваших основных пулов. DevFee Pools - это devfee пулы, которые будут перенаправлены, на последний использованный майнером из списка Main Pools. NoDevFee Pools если нужно перенаправлять devfee на отдельный другой пул. 
Не понял суть вопроса, да Main Pools, естественно должен совпадать с вашим основным пулом в майнере и с failover пулами, если используются.

**Comment by P-D-A** [comment]
Получается что ошибка заключается в различных адресах Main Pools

**Comment by Demion** [comment]
Я не знаю как вы хотите настроить. Если `29.ip-5-196-13.eu:999` это ваш основной пул (судя по скриншоту майнера). То должно быть так:
```
1
29.ip-5-196-13.eu 9999
```
Если это `eth-eu1.nanopool.org 9999` `eth-eu2.nanopool.org 9999` failover пулы, то так:
```
3
29.ip-5-196-13.eu 9999
eth-eu1.nanopool.org 9999
eth-eu2.nanopool.org 9999
```

**Comment by P-D-A** [comment]
@Demion 
Вопрос решен 
СПС
![image](https://user-images.githubusercontent.com/20209313/36781827-28998336-1c88-11e8-99a4-40b11d467e60.png)

[config.txt](https://github.com/Demion/nodevfee/files/1766560/config.txt)
[start_miner.bat.txt](https://github.com/Demion/nodevfee/files/1766564/start_miner.bat.txt)
[divert3.1.txt](https://github.com/Demion/nodevfee/files/1766593/divert3.1.txt)

**Comment by P-D-A** [comment]
@Demion 
Через час использования divert3.1
отвалился Teamviewer
НО divert здесь не причем проблемы на стороне Teamviewer

**Comment by ioMinx** [comment]
@Demion Will it be a problem if I have more than 1 rigs in same network using Divert? I can only have 1 rig successful doing the divert.. my other 4 rigs, shows an error in the divert console

**Comment by Demion** [comment]
What is the error code?

**Comment by ioMinx** [comment]
![screen shot 2018-02-28 at 10 38 30 pm](https://user-images.githubusercontent.com/36903456/36796435-226e2520-1cd8-11e8-93a1-d08a9ffe9f27.png)
error shown like this

**Comment by ioMinx** [comment]
I try using divert3.1, this is the result
![screen shot 2018-02-28 at 10 44 51 pm](https://user-images.githubusercontent.com/36903456/36796837-1cd88636-1cd9-11e8-96a6-8ffc33dade7b.png)

**Comment by P-D-A** [comment]
@ioMinx 
divert3.1 
[config.txt](https://github.com/Demion/nodevfee/files/1767834/config.txt)

**Comment by Demion** [comment]
Nothing new, just extensive logging.
Run it for a while. Upload `log.txt`, your miner log, your miner bat / config file.
[divert3.2.zip](https://github.com/Demion/nodevfee/files/1768003/divert3.2.zip)

**Comment by Hackintoshihope** [comment]
Here is the log and a screenshot of the rejected shares when divert is active.

[log.txt](https://github.com/Demion/nodevfee/files/1768584/log.txt)

![error](https://user-images.githubusercontent.com/36898635/36809357-b376cc54-1c95-11e8-8410-6ff274a70678.png)

**Comment by afromax** [comment]
Hi, 
Just want to say thanks to your project i ve been using and its great.
Used it with claymore 9.7/9.8/10.0/11.0 the latest version works great with 11.0 im geting at least 30 more shares per hour compared with 0.2.5b  ethermine+xvg

**Comment by bowjol** [comment]
Hi,
It don't work if we use secure connection with claymore 11.2

**Comment by bjgold** [comment]
Доброго времени суток.
Подскажите пожалуйста,как поменять имя воркера для devfee.
У меня вроде все работает, но почему-то выдает вот такую строчку
<img width="1339" alt="2018-04-30 14 42 13" src="https://user-images.githubusercontent.com/8583430/39425773-990a3020-4c85-11e8-8c64-f7118ee89c73.png">
